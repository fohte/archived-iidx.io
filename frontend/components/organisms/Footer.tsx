import { Icon } from 'antd'

const originalCopyrightYear = 2018

const getCopyrightYears = (): string => {
  const currentYear = new Date().getFullYear()

  if (currentYear === originalCopyrightYear) {
    return `${originalCopyrightYear}`
  }

  return `${originalCopyrightYear}-${currentYear}`
}

const Footer: React.SFC = () => (
  <>
    <Icon type="copyright" /> {getCopyrightYears()} iidx.io
  </>
)

export default Footer
