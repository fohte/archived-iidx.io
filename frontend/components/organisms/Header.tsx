import { Menu } from 'antd'
import Link from 'next/link'
import styled from 'styled-components'

import Brand from '@app/components/atoms/Brand'
import UserNav from '@app/components/molecules/UserNav'
import withAuthState from '@app/lib/withAuthState'

const Header: React.SFC = () => (
  <Menu theme="dark" style={{ lineHeight: '64px' }}>
    <Link href="/" prefetch>
      <BrandWrapper>
        <Brand />
      </BrandWrapper>
    </Link>
    <UserNavWrapper>
      <EnhancedUserNav />
    </UserNavWrapper>
  </Menu>
)

const BrandWrapper = styled.div`
  float: left;
`

const UserNavWrapper = styled.div`
  float: right;
`

const EnhancedUserNav = withAuthState()(UserNav)

export default Header
