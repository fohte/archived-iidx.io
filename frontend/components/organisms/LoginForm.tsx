import { Button, Form, Icon, Input } from 'antd'
import { FormComponentProps } from 'antd/lib/form'
import * as React from 'react'

import InputIcon from '@app/components/atoms/InputIcon'
import withSubmitHandling, {
  ComponentExternalProps,
  InjectedProps,
} from '@app/lib/withSubmitHandling'

export type ExternalProps = ComponentExternalProps<FormValues>

export interface Props
  extends ExternalProps,
    FormComponentProps,
    InjectedProps {}

export interface State {
  submitting: boolean
  errorMessage: string | null
}

export interface FormValues {
  email: string
  password: string
}

const LoginForm: React.SFC<Props> = ({
  form: { getFieldDecorator },
  errorMessage,
  handleSubmit,
  submitting,
}) => (
  <Form onSubmit={handleSubmit}>
    <Form.Item>
      {getFieldDecorator('email', {
        rules: [{ required: true, message: 'Please input your email.' }],
      })(<Input prefix={<InputIcon type="mail" />} placeholder="Email" />)}
    </Form.Item>
    <Form.Item>
      {getFieldDecorator('password', {
        rules: [{ required: true, message: 'Please input your password.' }],
      })(
        <Input
          prefix={<InputIcon type="lock" />}
          type="password"
          placeholder="Password"
        />,
      )}
    </Form.Item>
    <Form.Item>
      <Button type="primary" htmlType="submit">
        {submitting ? <Icon type="loading" /> : 'Log in'}
      </Button>
      {errorMessage}
    </Form.Item>
  </Form>
)

const WrappedNormalLoginForm = Form.create<ExternalProps>()(
  withSubmitHandling<FormValues>()(LoginForm),
)

export default WrappedNormalLoginForm
