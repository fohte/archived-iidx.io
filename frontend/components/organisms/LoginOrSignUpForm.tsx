import * as React from 'react'
import { Field as FinalField, Form as FinalForm } from 'react-final-form'
import {
  Button,
  Form,
  Icon,
  Input,
  Label,
  Message,
  Segment,
} from 'semantic-ui-react'
import { isEmail, isEmpty } from 'validator'

import withSubmitHandling, {
  ComponentExternalProps,
  InjectedProps,
} from '@app/lib/withSubmitHandling'

export interface ExternalProps extends ComponentExternalProps<FormValues> {
  submitText: string
}

export interface Props extends ExternalProps, InjectedProps {}

export interface FormValues {
  email: string
  password: string
}

const validators: {
  [key in keyof FormValues]: (
    value: FormValues[key] | null,
    allValues: FormValues,
  ) => string | undefined
} = {
  email: (value, _) => {
    if (!value || isEmpty(value)) {
      return 'Please input your email.'
    }

    if (!isEmail(value)) {
      return 'Email is invalid.'
    }
  },
  password: (value, _) => {
    if (!value || isEmpty(value)) {
      return 'Please input your password.'
    }
  },
}

const LoginOrSignUpForm: React.SFC<Props> = ({ handleSubmit, submitText }) => (
  <FinalForm
    onSubmit={handleSubmit}
    initialValues={{ email: '', password: '' }}
  >
    {({
      handleSubmit: innerHandleSubmit,
      pristine,
      hasValidationErrors,
      hasSubmitErrors,
      submitting,
      submitError,
    }) => (
      <Form size="large" onSubmit={innerHandleSubmit} error={hasSubmitErrors}>
        <Segment stacked textAlign="left">
          <FinalField name="email" validate={validators.email}>
            {({ input, meta }) => (
              <Form.Field error={!!(meta.touched && meta.error)}>
                <Input
                  fluid
                  type="email"
                  icon="user"
                  iconPosition="left"
                  {...input}
                  placeholder="E-mail address"
                />
                {meta.touched &&
                  meta.error && (
                    <Label basic color="red" pointing>
                      {meta.error}
                    </Label>
                  )}
              </Form.Field>
            )}
          </FinalField>
          <FinalField name="password" validate={validators.password}>
            {({ input, meta }) => (
              <Form.Field error={!!(meta.touched && meta.error)}>
                <Input
                  fluid
                  type="password"
                  icon="lock"
                  iconPosition="left"
                  {...input}
                  placeholder="Password"
                />
                {meta.touched &&
                  meta.error && (
                    <Label basic color="red" pointing>
                      {meta.error}
                    </Label>
                  )}
              </Form.Field>
            )}
          </FinalField>
          {hasSubmitErrors && <Message error content={submitError} />}
          <Button
            type="submit"
            disabled={submitting || pristine || hasValidationErrors}
            color="teal"
            fluid
            size="large"
          >
            {submitting ? <Icon loading name="spinner" /> : submitText}
          </Button>
        </Segment>
      </Form>
    )}
  </FinalForm>
)

const WrappedNormalLoginOrSignUpForm = withSubmitHandling<FormValues>()(
  LoginOrSignUpForm,
)

export default WrappedNormalLoginOrSignUpForm
