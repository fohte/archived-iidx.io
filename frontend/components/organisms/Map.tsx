import * as React from 'react'
import { Segment } from 'semantic-ui-react'

import MapTitleHeader from '@app/components/molecules/MapTitleHeader'
import ResultTable, { Result } from '@app/components/molecules/ResultTable'
import { FindMapMap, FindMapMusic } from '@app/queries'

export type Props = {
  music: FindMapMusic
  map: FindMapMap
  result?: Result
}

const Map: React.SFC<Props> = ({ music, map, result }) => (
  <>
    <Segment basic textAlign="center">
      <MapTitleHeader music={music} map={map} />
    </Segment>

    <Segment basic>
      <ResultTable map={map} result={result} />
    </Segment>
  </>
)

export default Map
