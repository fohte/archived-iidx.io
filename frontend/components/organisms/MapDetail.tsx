import * as React from 'react'
import { Segment } from 'semantic-ui-react'

import MapTitleHeader from '@app/components/molecules/MapTitleHeader'
import ResultTable, { Result } from '@app/components/molecules/ResultTable'
import { FindMapMap, FindMapMusic } from '@app/queries'

export type Props = {
  music: FindMapMusic
  map: FindMapMap
  result?: Result
}

const MapDetail: React.SFC<Props> = ({ music, map, result }) => (
  <>
    <Segment basic textAlign="center">
      <MapTitleHeader music={music} map={map} />
    </Segment>

    <Segment basic>
      <ResultTable maps={[{ ...map, result }]} />
    </Segment>
  </>
)

export default MapDetail
