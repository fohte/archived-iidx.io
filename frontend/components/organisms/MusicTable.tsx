import { GetMusicsWithMapsMusics } from '@app/queries'
import { Table } from 'antd'
import { ColumnProps } from 'antd/lib/table'
import flatMap from 'lodash/flatMap'

interface Props {
  musics: GetMusicsWithMapsMusics[]
}

interface Data {
  key: string
  title: string
  difficulty: string
  level: number
}

const columns: Array<ColumnProps<Data>> = [
  {
    title: 'Title',
    key: 'title',
    dataIndex: 'title',
  },
  {
    title: 'Difficulty',
    key: 'difficulty',
    dataIndex: 'difficulty',
  },
  {
    title: 'Level',
    key: 'level',
    dataIndex: 'level',
  },
]

const toTableData = (musics: Props['musics']): Data[] =>
  flatMap(musics, ({ title, subTitle, maps }) =>
    maps.map(({ id, difficulty, level }) => ({
      key: id,
      title: `${title} ${subTitle}`,
      difficulty,
      level,
    })),
  )

const MusicTable: React.SFC<Props> = ({ musics }) => {
  return <Table columns={columns} dataSource={toTableData(musics)} />
}

export default MusicTable
