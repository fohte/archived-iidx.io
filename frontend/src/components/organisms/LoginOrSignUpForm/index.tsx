import * as React from 'react'
import { Field as FinalField, Form as FinalForm } from 'react-final-form'
import { isEmail, isEmpty } from 'validator'

import Alert from '@app/components/atoms/Alert'
import Button from '@app/components/atoms/Button'
import FormGroup from '@app/components/atoms/FormGroup'
import InputText from '@app/components/atoms/InputText'
import withSubmitHandling, {
  ComponentExternalProps,
  InjectedProps,
} from '@app/lib/withSubmitHandling'

export interface ExternalProps extends ComponentExternalProps<FormValues> {
  submitText: string
}

export interface Props extends ExternalProps, InjectedProps {}

export interface FormValues {
  email: string
  password: string
}

const validators: {
  [key in keyof FormValues]: (
    value: FormValues[key] | null,
    allValues: FormValues,
  ) => string | undefined
} = {
  email: (value, _) => {
    if (!value || isEmpty(value)) {
      return 'Please input your email.'
    }

    if (!isEmail(value)) {
      return 'Email is invalid.'
    }
  },
  password: (value, _) => {
    if (!value || isEmpty(value)) {
      return 'Please input your password.'
    }
  },
}

const LoginOrSignUpForm: React.SFC<Props> = ({ handleSubmit, submitText }) => (
  <FinalForm
    onSubmit={handleSubmit}
    initialValues={{ email: '', password: '' }}
  >
    {({
      handleSubmit: innerHandleSubmit,
      pristine,
      hasValidationErrors,
      hasSubmitErrors,
      submitting,
      submitError,
    }) => (
      <form onSubmit={innerHandleSubmit}>
        <FinalField name="email" validate={validators.email}>
          {({ input, meta }) => (
            <FormGroup
              label="E-mail address"
              error={!!(meta.touched && meta.error)}
              errorMessage={meta.error}
            >
              <InputText
                type="email"
                error={!!(meta.touched && meta.error)}
                {...input}
                placeholder="E-mail address"
              />
            </FormGroup>
          )}
        </FinalField>
        <FinalField name="password" validate={validators.password}>
          {({ input, meta }) => (
            <FormGroup
              label="Password"
              error={!!(meta.touched && meta.error)}
              errorMessage={meta.error}
            >
              <InputText
                type="password"
                error={!!(meta.touched && meta.error)}
                {...input}
                placeholder="Password"
              />
            </FormGroup>
          )}
        </FinalField>
        {hasSubmitErrors && <Alert>{submitError}</Alert>}
        <Button
          type="submit"
          disabled={submitting || pristine || hasValidationErrors}
          loading={submitting}
        >
          {submitText}
        </Button>
      </form>
    )}
  </FinalForm>
)

const WrappedNormalLoginOrSignUpForm = withSubmitHandling<FormValues>()(
  LoginOrSignUpForm,
)

export default WrappedNormalLoginOrSignUpForm
