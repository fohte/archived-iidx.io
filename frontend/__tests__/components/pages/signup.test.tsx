import { mount, ReactWrapper } from 'enzyme'
import * as _ from 'lodash'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

import { FormValues } from '@app/components/organisms/LoginOrSignUpForm'
import Signup from '@app/pages/signup'

const mockCreate = jest.fn()

jest.mock('@app/lib/firebaseApp', () => ({
  auth: jest.fn().mockImplementation(() => ({
    createUserWithEmailAndPassword: mockCreate,
  })),
}))

const inputFields = (wrapper: ReactWrapper, values: FormValues) => {
  _.forEach(values, (value, key) => {
    wrapper
      .find(`input[name="${key}"]`)
      .simulate('change', { target: { value } })
  })
}

const submit = (wrapper: ReactWrapper) => {
  wrapper.find('button[type="submit"]').simulate('submit')
}

describe('/signup', () => {
  beforeEach(() => {
    mockCreate.mockClear()
  })

  it('create a user on firebase auth', () => {
    const wrapper = mount(<Signup />)

    inputFields(wrapper, { email: 'email@example.com', password: 'password' })
    submit(wrapper)

    expect(mockCreate).toHaveBeenCalledWith('email@example.com', 'password')
  })

  it('renders correctly', () => {
    const component = renderer.create(<Signup />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
