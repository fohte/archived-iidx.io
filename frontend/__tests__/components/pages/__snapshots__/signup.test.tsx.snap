// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/signup renders correctly 1`] = `
<div
  className="box"
>
  <nav
    className="headerNav"
  >
    <div
      className="container "
    >
      <div
        className="flexContainer"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <div
            className="item"
          >
            <span>
              iidx.io
            </span>
          </div>
        </a>
        <div
          className="item"
        >
          loading
        </div>
      </div>
    </div>
  </nav>
  <div
    className="content centering"
  >
    <div
      className="container"
    >
      <div
        className="box transparent header"
      >
        <h2>
          Create your personal account.
        </h2>
      </div>
      <div
        className="box "
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              className="label "
            >
              E-mail address
            </label>
            <div
              className="input-area "
            >
              <input
                className="input-text "
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="E-mail address"
                value=""
              />
            </div>
          </div>
          <div
            className="form-group"
          >
            <label
              className="label "
            >
              Password
            </label>
            <div
              className="input-area "
            >
              <input
                className="input-text "
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Password"
                value=""
              />
            </div>
          </div>
          <button
            className="button disabled expand "
            disabled={true}
            type="submit"
          >
            Sign up
          </button>
        </form>
      </div>
      <div
        className="box transparent footer"
      >
        Already have an account?
         
        <a
          href="/login"
          onClick={[Function]}
        >
          Log in.
        </a>
      </div>
    </div>
  </div>
  <footer
    className="box"
  >
    <div
      className="container "
    >
      <div
        className="inner"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-copyright fa-w-16 "
          data-icon="copyright"
          data-prefix="far"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm0 448c-110.532 0-200-89.451-200-200 0-110.531 89.451-200 200-200 110.532 0 200 89.451 200 200 0 110.532-89.451 200-200 200zm107.351-101.064c-9.614 9.712-45.53 41.396-104.065 41.396-82.43 0-140.484-61.425-140.484-141.567 0-79.152 60.275-139.401 139.762-139.401 55.531 0 88.738 26.62 97.593 34.779a11.965 11.965 0 0 1 1.936 15.322l-18.155 28.113c-3.841 5.95-11.966 7.282-17.499 2.921-8.595-6.776-31.814-22.538-61.708-22.538-48.303 0-77.916 35.33-77.916 80.082 0 41.589 26.888 83.692 78.277 83.692 32.657 0 56.843-19.039 65.726-27.225 5.27-4.857 13.596-4.039 17.82 1.738l19.865 27.17a11.947 11.947 0 0 1-1.152 15.518z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        2018
         iidx.io |
         
        test
      </div>
    </div>
  </footer>
</div>
`;
